webapp/packages/plugin-data-viewer/src/DatabaseDataModel/Actions/ResultSet/ResultSetDataContentAction.ts

```diff
@@ -165,69 +149,81 @@ export class ResultSetDataContentAction extends DatabaseDataAction<any, IDatabas
 
     const url = await this.source.runTask(async () => {
       try {
-        this.activeElement = element;
-        const fileName = await this.loadFileName(this.result, column.position, row);
-        return this.generateFileDataUrl(fileName);
+        this.updateCache(element, { loading: true });
+        return await this.loadDataURL(this.result, column.position, row);
       } finally {
-        this.activeElement = null;
+        this.updateCache(element, { loading: false });
       }
     });
 
     return url;
   }
 
   async resolveFileDataUrl(element: IResultSetElementKey) {
-    const cachedUrl = this.retrieveFileDataUrlFromCache(element);
+    const cachedUrl = this.retrieveBlobFromCache(element);
 
     if (cachedUrl) {
       return cachedUrl;
     }
 
     const url = await this.getFileDataUrl(element);
```

Comment:
this function sets cache loading state as expects
but we have a somekind of freeze in view button cause const `blob = await downloadFromURL(url);` below is not handled with loader, I suppose

I've send you the video

Reply:
will be fixed later